msig_contract_deployment/msig_erc20_bootstrap.md

```diff
@@ -0,0 +1,39 @@
+
+Action 1: Update the existing evm_runtime contract, adding supports for the following features required for ERC20 bridging:
+  - *call action: this action will allow any account in the EOS side acts as the corresponding account (using deterministric reserverd address) in EVM side*
+  - *admincall action: this action allow ENF & BPs to co-sign to satisfy eosio.evm@active and act on behalf of any EVM account in emergency situations.*
+  - *event mechanisim: this is the mechanisim allowing event generated in the solidity contract in EVM to be sent as bridge message via inline action in native EOS.*
+
+Action 2: withdraw 300 EOS from the eosio.evm's fee balance to enf for covering RAM and as ERC-20 bridge funds 
+
+## Multi-sig 2: ERC-20 contract bootstrapping (depends on Multi-sig 1):
+
+Action 1: Deploy core EVM ERC-20 contract to account `eosio.erc2o`. This contract contains the following actions:
+  - *upgrade, upgradeto: set or deploy the ERC-20 upgradable implementation contract into EVM side and register the contract address in eosio.erc2o's multi_index table in EOS side.*
+  - *regtoken: deploy the new proxy ERC-20 contract on EVM side (which will call the upgradable implementation contract) and setup the link between existing EOS tokens (such as USDT, IQ, ... etc) and the EVM proxy contract.*
+  - *transfer notify handler (EOS->EVM flow implementation): action handler to call the EVM's ERC-20 token mint action (with the memo identifying the destination EVM address) via the eosio.evm contract, upon receiving the correct registered tokens from the EOS side.*
+  - *onbridgemsg: event handler (EVM->EOS flow implmenetation): action hanlder to call the EOS inline transfer to transfer the registered token to the destination EOS account upon receiving the EVM event which is genereted from the implementation contract and sent via the eosio.evm runtime contract.*
+  - *addregress, removeegress: similar to native EOS bridge, it controls the whilelist of EOS destinaction addresses with deployed code, to prevent misuse of CPU resource.*
+  - *setegressfee, setingressfee: actions to control the bridge fee. Ingressfee is charged in form of the native tokens, while egressfee is charged in form on EOS.*
+  - *withdrawfee: action to withdraw the accumulated ingress fee*
+
+Action 2: add code permission to `eosio.erc2o`
+
+Action 3: Deploy the new deposit proxy contract to the `eosio.evmin` account. It acts as a proxy in the EOS->EVM flow to receive token and redirect received tokens to proper accounts (either eosio.evm for native EOS token, or eosio.erc2o for other tokens)
+
+Action 4: add code permission to `eosio.evmin`
+
+Action 5: Call bridgereg in EOS EVM Contract for the eosio.erc2o receiver (this also opens the balance for eosio.erc2o). Use 0.01 EOS as minimum bridge fee.
+
+Action 6: Transfer 100 EOS to eosio.erc2o account in the EVM side as initial bridging funds
+
+Action 7: Call upgradeto action on the erc20 contract to initialize it.
```

Comment:
Might be worth point out here that on testnet the contract address is `0x7F7cbF0f0f62bC0bb8b43823B8Eaf1DBe8f10E68` while on mainnet the contract address is `0x6446D14972485B67b852fB3fC532bc2B854f3c14`.